### ISSUE TITLE
Footprint name clamping at 32 chars generates Exception in case of similar names of footprints

### ISSUE THREAD

**Issue by francescocaponio** [issus/AltiumSharp#18]
I noticed an issue when trying to import footprints with more than 32 chars and a similar name up until the 32nd char.

The problem appears in the function `ReadSectionKeys` of `PcbLibReader.cs`, and both `libRef` and `sectionKey` appear to clamp the name at 32 chars. The first item is added to the SectionKeys List, while the second one, which shows the same libRef, is not added and triggers the exception.

The debugger shows the Exception System.ArgumentException: Another element with the same key has been already added.

I attached a picture with the result and the same pcblib file used for the screenshot with just two footprints with similar names which will prompt the issue.

![image](https://user-images.githubusercontent.com/66728654/194921286-4ecab504-979a-45cf-8851-ea56dac4cd64.png)
[PcbLib3.zip](https://github.com/issus/AltiumSharp/files/9747862/PcbLib3.zip)

**Comment by francescocaponio** [comment]
So, I went a little bit more deep in the file format. I think that the string reader for this section (SectionKeys) would be different from the string reader used in other places in the same PCB lib.

![image](https://user-images.githubusercontent.com/66728654/195195954-d52bef22-4c47-4c2c-9b05-922ed01b02c5.png)

Take a look at the attached screenshot. It is the raw stream from the previous library for the SectionKeys stream.

The Int32 data read at offset 4 is the lenght of the following block, but the following 0x20 is not a blankspace, it is the string lenght instead, similarly to what the do in the raw Pin representation on the Schlib for the pin name, designator and Swap ID.

To summarize, the problems I see are two.

- a raw char indicating the string lenght is taken as part of the string for both the libref and sectionkey read from this stream (in the beginning of the string).
- the last char is dropped because it's taken as the string is 0 terminated while it is not.

I still didn't go as deep in your code to suggest how to edit it, or send you a PR yet, but I hope it was more clear to you what I see.

**Comment by francescocaponio** [comment]
What the library is doing at the moment is that it is starting reading the string at offset 8 (taking a whitespace in the beginning, you can notice that in the picture of the first comment) until offset 0x29 (reading 33 bytes) and then dropping the last one since it is assumed the string would be 0 terminated, leaving the string as `  VERY_LONG_USEFUL_FOOTPRINT_NAME` without the number 1 at the end, with a whitespace in the beginning of the string.
